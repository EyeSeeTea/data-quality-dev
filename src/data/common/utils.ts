import { MetadataItem } from "$/domain/entities/MetadataItem";
import { Id } from "$/domain/entities/Ref";
import { TrackedEntitiesGetResponse } from "$/types/d2-api";
import { TrackerEventsResponse } from "@eyeseetea/d2-api/api/trackerEvents";

export function getProgramStageIndexById(programStageId: Id, metadata: MetadataItem): number {
    const programStageIndex = metadata.programs.qualityIssues.programStages.findIndex(
        programStage => programStage.id === programStageId
    );
    if (programStageIndex === -1) throw Error(`Cannot found programStage: ${programStageId}`);
    return programStageIndex;
}

export function buildTrackerResponse(
    response: TrackedEntitiesGetResponse & {
        trackedEntities?: TrackedEntitiesGetResponse["instances"];
    }
): TrackedEntitiesGetResponse {
    if (!response.instances && response.trackedEntities) {
        return { ...response, instances: response.trackedEntities };
    } else if (!response.trackedEntities && response.instances) {
        return response;
    } else {
        return response;
    }
}

export function buildTrackerEventsResponse(
    response: TrackerEventsResponse & {
        events?: TrackerEventsResponse["instances"];
    }
): TrackerEventsResponse {
    if (!response.instances && response.events) {
        return { ...response, instances: response.events };
    } else if (!response.events && response.instances) {
        return response;
    } else {
        return response;
    }
}
