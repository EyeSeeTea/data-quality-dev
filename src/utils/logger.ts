import { TrackerProgramLogger, initLogger } from "@eyeseetea/d2-logger";
import { Id } from "../domain/entities/Ref";

export let logger: TrackerProgramLogger;

export async function setupLogger(
    baseUrl: string,
    trackerProgramId: Id,
    options?: { isDebug?: boolean }
): Promise<void> {
    const { isDebug = false } = options ?? {};

    logger = await initLogger({
        type: "trackerProgram",
        debug: isDebug,
        baseUrl: baseUrl,
        auth: undefined,
        trackerProgramId: trackerProgramId,
        messageTypeId: undefined,
    });

    if (!logger) throw Error(`Failed to initialise the logger.`);
}
