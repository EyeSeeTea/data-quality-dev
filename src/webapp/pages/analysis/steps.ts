import i18n from "../../../utils/i18n";
import { ConfigurationStep } from "./steps/ConfigurationStep";
import { DensityStep } from "./steps/DensityStep";
import { DisaggregatesStep } from "./steps/DisaggregatesStep";
import { DoctorsStep } from "./steps/DoctorsStep";
import { MidwiferyStep } from "./steps/MidwiferyStep";
import { NursingMidwiferyStep } from "./steps/NursingMidwiferyStep";
import { NursingStep } from "./steps/NursingStep";
import { OtherStep } from "./steps/OtherStep";
import { OutliersStep } from "./steps/OutliersStep";
import { TrendsStep } from "./steps/TrendsStep";

export const steps = [
    {
        key: "configuration",
        label: i18n.t("Configuration"),
        component: ConfigurationStep,
    },
    {
        key: "outliers",
        label: i18n.t("Outliers"),
        component: OutliersStep,
    },
    {
        key: "trends",
        label: i18n.t("Trends"),
        component: TrendsStep,
    },
    {
        key: "disaggregates",
        label: i18n.t("Disaggregates"),
        component: DisaggregatesStep,
    },
    {
        key: "doctors",
        label: i18n.t("Doctors"),
        component: DoctorsStep,
    },
    {
        key: "nursing",
        label: i18n.t("Nursing"),
        component: NursingStep,
    },
    {
        key: "midwifery",
        label: i18n.t("Midwifery"),
        component: MidwiferyStep,
    },
    {
        key: "nursing-midwifery",
        label: i18n.t("Nursing/Midwifery"),
        component: NursingMidwiferyStep,
    },
    {
        key: "density",
        label: i18n.t("Density"),
        component: DensityStep,
    },
    {
        key: "other",
        label: i18n.t("Other"),
        component: OtherStep,
    },
];
