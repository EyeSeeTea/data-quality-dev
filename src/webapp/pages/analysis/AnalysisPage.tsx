import React from "react";
import { Wizard } from "@eyeseetea/d2-ui-components";
import { PageHeader } from "$/webapp/components/page-header/PageHeader";
import i18n from "$/utils/i18n";
import { steps } from "./steps";

type PageProps = {
    name: string;
};

export const AnalysisPage: React.FC<PageProps> = React.memo(props => {
    const { name } = props;
    return (
        <React.Fragment>
            <PageHeader title={i18n.t(name)} />
            <Wizard steps={steps} />
        </React.Fragment>
    );
});
