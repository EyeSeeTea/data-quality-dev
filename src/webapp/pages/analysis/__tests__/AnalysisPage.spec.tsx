import { getReactComponent } from "$/utils/tests";
import { AnalysisPage } from "$/webapp/pages/analysis/AnalysisPage";

describe("AnalysisPage", () => {
    it("renders the feedback component", async () => {
        const view = getView();

        expect(await view.findByText("Analysis")).toBeInTheDocument();
        expect(view.asFragment()).toMatchSnapshot();
    });
});

function getView() {
    return getReactComponent(<AnalysisPage name="Analysis" />);
}
