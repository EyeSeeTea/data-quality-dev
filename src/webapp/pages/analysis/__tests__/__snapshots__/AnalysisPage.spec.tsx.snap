// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnalysisPage > renders the feedback component 1`] = `
<DocumentFragment>
  <div>
    <h5
      class="MuiTypography-root sc-bczRLJ WGVdy MuiTypography-h5 MuiTypography-gutterBottom"
      data-test="page-header-title"
    >
      Analysis
    </h5>
  </div>
  <div
    class=" makeStyles-root-10"
  >
    <div
      class="MuiPaper-root MuiStepper-root MuiStepper-horizontal Wizard-Stepper makeStyles-stepper-15 MuiPaper-elevation0"
    >
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal current-step"
          data-test-current="true"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  1
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
              >
                Configuration
              </span>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root MuiStepButton-touchRipple"
          />
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Outliers
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  3
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Trends
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  4
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Disaggregates
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  5
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Doctors
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  6
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Nursing
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  7
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Midwifery
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  8
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Nursing/Midwifery
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  9
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Density
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal Wizard-Step"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root makeStyles-stepButton-14 MuiStepButton-horizontal Mui-disabled"
          data-test-current="false"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  10
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                Other
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-contents-13 MuiPaper-elevation1 MuiPaper-rounded"
      data-wizard-contents="true"
    >
      <div>
        <h2>
          Configuration
        </h2>
      </div>
      <div
        class="Wizard-Navigation makeStyles-buttonContainer-18"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-16 Mui-disabled makeStyles-buttonDisabled-17 Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            ← Previous
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-16"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next →
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
